app/views/directives/create-secret.html

```diff
@@ -47,268 +47,296 @@
-          Password or token is required.
-        </div>
-      </div>
-      <div class="help-block" id="password-token-help">
-        Password or token for Git authentication. Required if a ca.crt or .gitconfig file is not specified.
-      </div>
-
-      <div class="form-group">
-        <div class="checkbox">
-          <label>
-            <input type="checkbox" ng-model="add.cacert">
-            Use a custom ca.crt file
-          </label>
+            aria-describedby="username-help"
+            required="true">
+          <span class="input-group-btn">
+            <button type="button" class="btn btn-default" ng-click="generateWebhookSecretKey()">Generate</button>
+          </span>
         </div>
-      </div>
-      <div class="form-group" ng-if="add.cacert" id="cacert">
-        <label class="required" for="cacert">CA Certificate File</label>
-        <osc-file-input
-          id="cacert-file-input"
-          model="newSecret.data.cacert"
-          drop-zone-id="cacert"
-          help-text="Upload your ca.crt file."
-          required="true"></osc-file-input>
-        <div ui-ace="{
-          mode: 'txt',
-          theme: 'eclipse',
-          rendererOptions: {
-            fadeFoldWidgets: true,
-            showPrintMargin: false
-          }
-        }" ng-model="newSecret.data.cacert" class="create-secret-editor ace-bordered" id="cacert-editor" required></div>
-      </div>
-
-
-    </div>
-
-    <div ng-if="newSecret.authType === 'kubernetes.io/ssh-auth'">
-      <div class="form-group" id="private-key">
-        <label for="privateKey" class="required">SSH Private Key</label>
-        <osc-file-input
-          id="private-key-file-input"
-          model="newSecret.data.privateKey"
-          drop-zone-id="private-key"
-          help-text="Upload your private SSH key file."></osc-file-input>
-        <div ui-ace="{
-          theme: 'eclipse',
-          rendererOptions: {
-            fadeFoldWidgets: true,
-            showPrintMargin: false
-          }
-        }" ng-model="newSecret.data.privateKey" class="create-secret-editor ace-bordered" id="private-key-editor" required></div>
-        <div class="help-block">
-          Private SSH key file for Git authentication.
+        <div class="help-block" id="secret-name-help">
+          Value of the secret will be supplied when invoking the webhook.
+          <a href="{{'webhooks' | helpLink}}" target="_blank"><span class="learn-more-inline">Learn More&nbsp;<i class="fa fa-external-link" aria-hidden="true"></i></span></a>
         </div>
       </div>
     </div>
 
-    <div ng-if="newSecret.type === 'source'">
+    <div ng-if="newSecret.type != 'webhook'">
       <div class="form-group">
-        <div class="checkbox">
-          <label>
-            <input type="checkbox" ng-model="add.gitconfig">
-            Use a custom .gitconfig file
-          </label>
-        </div>
+        <label for="authentificationType">Authentication Type</label>
```

Comment:
I see you didn't create this, but can you fix it?  Change to kebab case and add an input-id to the ui-select with the same kebab cased value.  It won't actually work because of https://github.com/angular-ui/ui-select/issues/1912, but at least the code will be right once the bug is resolved.